runtime/bundles/org.eclipse.core.contenttype/src/org/eclipse/core/runtime/content/IContentType.java

```diff
@@ -167,26 +169,56 @@ public interface IContentType extends IContentTypeSettings {
 	String getName();
 
 	/**
-	 * Returns whether this content type is associated with the
-	 * given file name.
+	 * Returns whether this content type is associated with the given file name. The
+	 * association is done ignoring case "c"=="C"
 	 *
 	 * @param fileName the file name
-	 * @return <code>true</code> if this content type is associated with
-	 * the given file name, <code>false</code> otherwise
+	 * @return <code>true</code> if this content type is associated with the given
+	 *         file name, <code>false</code> otherwise
 	 * @see #isAssociatedWith(String, IScopeContext)
 	 */
+	@Deprecated
```

Comment:
If you deprecate it, there should be javadoc explaining what to use instead.

I wonder though, why deprecate it instead of having a default method that calls the new method with true, which has been the behavior of the past?  But then, you've deprecated two methods and replaced them with a single method, so if I have a string but not a scope context, what will I do to fix the deprecation warning?